### DISCUSSION TITLE
SDXL-Lightning by ByteDance

### DISCUSSION THREAD

**Discussion by thiner** [lllyasviel/Fooocus#2318] [Ideas]
A new base model, https://huggingface.co/ByteDance/SDXL-Lightning just released by ByteDance. They claimed it can generate high quality images in just 2 steps(even 1 step).

**Comment by mashb1t** [comment]
@thiner thanks for the info, this should be usable OOTB with overriding the sampling steps as well as cfg scale, similar to sdxl turbo!

**Reply by newxhy** [reply]
https://huggingface.co/ByteDance/SDXL-Lightning，Forge is already supported, how to set fooocus?

**Reply by mashb1t** [reply]
As mentioned in my last comment, by overriding parameters. See https://github.com/lllyasviel/Fooocus/issues/1499 for more insights how turbo can be set up. I haven't tested lightning, but the setup should be similar with a few adjusted values you can find in their comfy flow/model page on HF.

**Reply by blckbx** [reply]
Been testing SDXL Lightning on Fooocus by importing the 8-step safetensor file and configuring CFG, steps, sampler and scheduler as described on huggingface.co. Works pretty good. Personally I prefer more steps for better quality over speedy generation. 

For reference:
```
/models/checkpoints/sdxl_lightning_8step.safetensors

Base Model: sdxl_lightning_8step.safetensors
Guidance Scale: 2
Sampler: Euler
Scheduler: sgm_uniform
Forced Overwrite of Sampling Step: 8
```

**Reply by Irmagon** [reply]
I try all combinations parameters...
This preset works fine:
```
"default_model": "sdxl_lightning_8step.safetensors",
"default_image_number": 1,
"default_cfg_scale": 1.0,
"default_sample_sharpness": 2.0,
"default_sampler": "dpmpp_2m_sde_gpu",
"default_scheduler": "lcm",
"default_performance": "Speed",
"default_overwrite_step": 8,

```

**Reply by mashb1t** [reply]
for me euler and sgm_uniform worked best, as they are more true to life (less bloom + saturation) and introduce less artifacts (e.g. on skin) than dpmpp_2m_sde_gpu and lcm

**Comment by xhoxye** [comment]
Some third-party lightning models can also be tried, such as the following:
https://civitai.com/models/112902/dreamshaper-xl?modelVersionId=354657
https://civitai.com/models/43977?modelVersionId=355837
https://civitai.com/models/129666?modelVersionId=356366

**Comment by thiner** [comment]
Is there any chance to refactor the **Extreme Speed** mode, switch from lcm to lightning, and reduce the steps to 4?

**Reply by mashb1t** [reply]
We have to keep in mind that there's a difference between models snd performance. Performance can be changed for every model independently, so Extreme Performance can be set for different models of all kinds. Lightning is one specific series of model differing in trained steps, but still fixed to steps. LCM is a speed LoRA.
=> lightning model can't replace the universally applicable performance Extreme Speed

**Reply by poor7** [reply]
> => lightning model can't replace the universally applicable performance Extreme Speed

@mashb1t  Lightning can. There are the same LoRAs, for example [this one for 8 steps](https://huggingface.co/ByteDance/SDXL-Lightning/resolve/main/sdxl_lightning_8step_lora.safetensors). I successfully use it with any model. 
Example image: 8 steps Juggernaut XL
![euler_ancestral](https://github.com/lllyasviel/Fooocus/assets/154618384/a6a8c34a-a3be-4863-802c-8838a34289fc)

**Reply by mashb1t** [reply]
@poor7 ah, ok, then i didn't have the full picture. I'll test this on the weekend.

**Reply by thiner** [reply]
@mashb1t Check this out: https://hf-mirror.com/ByteDance/SDXL-Lightning/resolve/main/sdxl_lightning_4step_lora.safetensors. They provide Lora models as well.

**Comment by kexul** [comment]
> > => lightning model can't replace the universally applicable performance Extreme Speed
> 
> @mashb1t Lightning can. There are the same LoRAs, for example [this one for 8 steps](https://huggingface.co/ByteDance/SDXL-Lightning/resolve/main/sdxl_lightning_8step_lora.safetensors). I successfully use it with any model. Example image: 8 steps Juggernaut XL ![euler_ancestral](https://private-user-images.githubusercontent.com/154618384/307264580-a6a8c34a-a3be-4863-802c-8838a34289fc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg4Nzg4ODksIm5iZiI6MTcwODg3ODU4OSwicGF0aCI6Ii8xNTQ2MTgzODQvMzA3MjY0NTgwLWE2YThjMzRhLWEzYmUtNDg2My04MDJjLTg4MzhhMzQyODlmYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIyNVQxNjI5NDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zNmNjMTZkM2Y1MmZlNTlmYTNjMjRiZGI1MjU3YTQzMTQ3MGU1MjczMjE1MGEwYmI5ZGYyODk0NGEwNDRjY2ZjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.apDH3XWEkUkvar79luCIW5T2VDNAtYwXH87264Ixbzw)

Would you mind share your setting, this is amazing!

**Comment by Green-li** [comment]
I have tried the 4 steps SDXL-Lightning LoRA on the fooocus default model and as well as the LCM on 4 steps.
Here are the images:
from left to right base model, +sdxl-lightning 4 steps lora, +lcm 4steps
![IMG_20240228_143414](https://github.com/lllyasviel/Fooocus/assets/11475982/1d37c663-d43b-4ea4-86d0-b4f43cfd1420)
![IMG_20240228_143506](https://github.com/lllyasviel/Fooocus/assets/11475982/5552f5c9-224f-411b-a446-7a14fd9f75e0)
![IMG_20240228_143539](https://github.com/lllyasviel/Fooocus/assets/11475982/4db4b86d-2583-4a5b-b20b-8bf6b29f3704)

**Reply by Green-li** [reply]
The LCM and SDXL-Lightning using 4 steps both cost 1s on my RTX 3060 GPU while the base model cost 22s using 30 steps.

**Comment by felixsanz** [comment]
I hope you don't consider this shameless spam... but I have written an article about SDXL-Lightning explaining how it works and made a lot of comparisons (steps, lora, vs turbo/lcmlora, etc).

I'm sure it will be useful!

https://www.felixsanz.dev/articles/sdxl-lightning-quick-look-and-comparison

**Reply by mashb1t** [reply]
Thank you for sharing, much appreciated!

**Reply by blckbx** [reply]
Wow great articles! Muchas gracias!

**Comment by mashb1t** [comment]
Please find the new performance mode proposal in https://github.com/lllyasviel/Fooocus/pull/2415.
Looking forward to your feedback!

**Reply by VelocityRa** [reply]
Thanks!
How to use finetunes like DreamShaper XL Lightning? It performs better according to [this](https://www.reddit.com/r/StableDiffusion/comments/1ax4col/i_compared_the_generation_results_of/) etc.
I made a json for it (latest release, not using your PR) but not sure I have the correct settings picked.

**Reply by mashb1t** [reply]
"Lightning" performance adds the LoRA for SDXL Lightning, which can be used with every base model.
DreamShaper XL Lightning and all "Lightning" models have this exact lightning LoRA already merged, same principle as for LCM.
Generally speaking you can use every Lightning model with Fooocus, but please keep before mentioned facts in mind and do not use performance lightning with them, as this will apply the Lightning LoRA twice and produces worse results.
Feel free to choose and use every model based on SDXL you like. You may have to adjust settings such as steps override though.

Hope this helps in understanding the structure ans background better.

**Reply by VelocityRa** [reply]
Thank you! What you said seems right from my tests.

For anyone looking for a preset json, here's one for Dreamshaper XL Lighting, just pass `--preset lightning` to Foocus (and keep the setting at `Speed`).
[lightning.json](https://github.com/lllyasviel/Fooocus/files/14628978/lightning.json)

**Reply by xhoxye** [reply]
lightning.json, it is recommended to change the name, if you use lightning, it may conflict with the 2.3.0 update

**Comment by BradKML** [comment]
@mashb1t thanks for all the work but is there SDXL-lightning settings for fast anime generation? Would like to see if it is viable on a laptop with embedded GPU

**Reply by mashb1t** [reply]
You can select the anime preset in combination with lightning performance
